src/main/java/mia/Command.java

```diff
@@ -0,0 +1,254 @@
+package mia;
+
+/**
+ * A {@code Command} is an actionable item that can be run by {@code Mia}.
+ *
+ * @author Richard Dominick
+ */
+abstract class Command {
+    private final Mia context;
+    private final boolean isExit;
+
+    private Command(Mia context) {
+        this(context, false);
+    }
+
+    private Command(Mia context, boolean isExit) {
+        this.context = context;
+        this.isExit = isExit;
+    }
+
+    /**
+     * Parses a user input string into a command that is appropriate for the calling context.
+     *
+     * @param context The context the command is intended to be executed from
+     * @param input The user input string to parse
+     * @return A {@code Command} for the context to call
+     * @throws IllegalArgumentException When a parsing error occurs
+     */
+    public static Command from(Mia context, String input) throws IllegalArgumentException {
```

Comment:
I don't see the need for having a separate enum for every single command type. I like to think of it as: what if in the future you want to have another command that exits while not saving any changes (for example)? What about other behaviour? It would not make sense to use multiple conditional checks for every single enum type.

But by using "flags" such as `shouldCloseContext`, or potentially `shouldXXXX` in the future, this makes it easier. Also using enums breaks abstraction. Why should a the `context` the command is being run in know about what types of `Command`s are available?